pkg/kubevirt/machine.go

```diff
@@ -46,40 +46,51 @@ type Machine struct {
 	getCommandExecutor func(string, *ssh.ClusterNodeSshKeys) ssh.VMCommandExecutor
 }
 
-// NewMachine returns a new Machine service for the given context.
-func NewMachine(ctx *context.MachineContext, client client.Client, namespace string, sshKeys *ssh.ClusterNodeSshKeys) (*Machine, error) {
+// NewBasicMachine returns a new Machine service for the given context, only with some basic fields.
+func NewBasicMachine(ctx *context.MachineContext, client client.Client, namespace string) (*Machine, error) {
```

Comment:
I don't think you need two functions here for the machine. passing `nil` for ssh keys should be fine, just make sure we have unit tests that exercise this situation during normal reconcile.